lib/galaxy/tools/parameters/dataset_matcher.py

```diff
@@ -121,8 +121,8 @@ def valid_hda_match(self, hda, check_implicit_conversions=True):
         else:
             if not check_implicit_conversions:
                 return False
-            target_ext, converted_dataset = hda.find_conversion_destination(formats)
-            if target_ext:
+            direct_match, target_ext, converted_dataset = hda.find_conversion_destination(formats)
```

Comment:
Same here, you can pull that up to line 119

Reply:
I think I covered all three comments .. please check my changes.